File: src/main/java/io/veridia/jsonlogic/helpers/ToBoolean.java
package io.veridia.jsonlogic.helpers;

import java.lang.reflect.Array;
import java.util.Collection;

public class ToBoolean {
    public static boolean eval(Object value) {
        if (value == null) return false;

        if (value instanceof Boolean) return (Boolean) value;

        if (value instanceof Number) {
            double doubleValue = ((Number) value).doubleValue();

            return !Double.isNaN(doubleValue) && doubleValue != 0;
        }

        if (value instanceof String) return !((String) value).isEmpty();

        if (value instanceof Collection) return !((Collection<?>) value).isEmpty();

        if (value.getClass().isArray())return Array.getLength(value) > 0;

        return true;
    }
}

File: src/main/scala/io/veridia/jsonlogic/helpers/ToBoolean.scala
package io.veridia.jsonlogic.helpers

// Truthiness rules derived from json-logic-java (MIT License) — see NOTICE.md.
object ToBoolean:
  def eval(value: Any): Boolean = value match
    case null => false
    case b: java.lang.Boolean => b.booleanValue()
    case n: Number =>
      val d = n.doubleValue()
      !d.isNaN && d != 0.0
    case s: String => s.nonEmpty
    case c: java.util.Collection[?] => !c.isEmpty
    case a if a.getClass.isArray => java.lang.reflect.Array.getLength(a) > 0
    case _ => true
